lib/ddtrace/contrib/rack/rum_injection.rb

```diff
@@ -0,0 +1,273 @@
+require 'ddtrace/contrib/rack/ext'
+require 'ddtrace/environment'
+require 'date'
+
+module Datadog
+  module Contrib
+    # Rack module includes middlewares that are required to trace any framework
+    # and application built on top of Rack.
+    module Rack
+      # RumInjection ensures that the Rack Response has rum information
+      # injected into it's html. The middleware modifies the response body
+      # of non-cached html so that it can be retrieved by the rum browser-sdk in
+      # the application frontend.
+      # rubocop:disable Metrics/ClassLength
+      class RumInjection
+        include Datadog::Environment::Helpers
+
+        RUM_INJECTION_FLAG = 'datadog.rum_injection_flag'.freeze
+        INLINE = 'inline'.freeze
+        IDENTITY = 'identity'.freeze
+        HTML_CONTENT = 'text/html'.freeze
+        XHTML_CONTENT = 'application/xhtml+xml'.freeze
+        NO_CACHE = 'no-cache'.freeze
+        NO_STORE = 'no-store'.freeze
+        PRIVATE = 'private'.freeze
+        MAX_AGE = 'max-age'.freeze
+        SMAX_AGE = 's-maxage'.freeze
+        MAX_AGE_ZERO = 'max-age=0'.freeze
+        SMAX_AGE_ZERO = 's-maxage=0'.freeze
+        CONTENT_TYPE_STREAMING = 'text/event-stream'.freeze
+        CACHE_CONTROL_HEADER = 'Cache-Control'.freeze
+        CONTENT_DISPOSITION_HEADER = 'Content-Disposition'.freeze
+        CONTENT_ENCODING_HEADER = 'Content-Encoding'.freeze
+        CONTENT_LENGTH_HEADER = 'Content-Length'.freeze
+        CONTENT_TYPE_HEADER = 'Content-Type'.freeze
+        EXPIRES_HEADER = 'Expires'.freeze
+        SURROGATE_CACHE_CONTROL_HEADER = 'Surrogate-Control'.freeze
+        TRANSFER_ENCODING_HEADER = 'Transfer-Encoding'.freeze
+        ACTION_CONTROLLER_INSTANCE = 'action_controller.instance'.freeze
+        TRANSFER_ENCODING_CHUNKED = 'chunked'.freeze
+
+        def initialize(app)
+          @app = app
+        end
+
+        def call(env)
+          result = @app.call(env)
+
+          begin
+            # TODO: env['rack.hijack?'] is true in rails tests
+            # dont check until we have more clarity on usage
```

Comment:
Let's remove this for now, as `rack.hijack?` only tells us that Rack itself supports hijacking.
It doesn't seem like we can detect a Rack hijack from the variable in the spec.
We should still be safe overriding the body in a spec-compliant way, which we are doing.